ext/opcache/Optimizer/dfa_pass.c

```diff
@@ -896,6 +898,7 @@ static int zend_dfa_optimize_jmps(zend_op_array *op_array, zend_ssa *ssa)
 					break;
 				}
 				case ZEND_SWITCH_LONG:
+				case ZEND_MATCH_LONG:
 					if (opline->op1_type == IS_CONST) {
 						zval *zv = CT_CONSTANT_EX(op_array, opline->op1.constant);
 						if (Z_TYPE_P(zv) != IS_LONG) {
```

Comment:
Same here - for ZEND_MATCH_LONG, I'd assume taking the last successor instead of the first successor would be more efficient.

I'm not familiar with what take_successor_ex is actually doing though. I'd hope that'd work, but it may end up being more complicated than that.

```
							take_successor_ex(ssa, block_num, block, block->successors[0]);  // take last successor instead
```